### ISSUE TITLE
Throw custom exception for WebView Control is no longer valid

### ISSUE THREAD

**Issue by RickStrahl** [MicrosoftEdge/WebView2Feedback#2254]
I'm seeing a lot of errors coming into my application with recent updates of the WebView control runtime.  For a bit things seemed to be much better, but the problem has returned and is more persistent.

![image](https://user-images.githubusercontent.com/1374013/157550977-e6859e4c-1a56-4421-b215-59c6623b3658.png)

One problem is that it's not easy to detect these failures. I have the `ProcessFailed` handler in place, but apparently these exceptions fire before that and they are fired ALOT (presumably on any WebView operation, but I don't know because the stack trace doesn't show anything from my code).

### Suggestion
One thing that would really help here if this could throw an easily trappable exception instead the generic `InvalidArgumentException`. Right now the only way I can do this is check if IVAE and then check the message which is not ideal. Basically when this happens I have to shut down the application. If not, there end up 100s of these errors flooding the logs in a matter of a couple of seconds before the Failed header kicks in or the app crashes (hard to tell as I've not seen it).

**Comment by champnic** [comment]
Hey @RickStrahl - thanks for the feature request, and sorry that these errors have started popping up more. I believe we decided to use the InvalidArgumentException with a custom message rather than a custom error type so that we wouldn't break app compat for apps that were already specifically catching the InvalidArgumentException.

Is this the same error you had mentioned in a different thread? The action we would likely take is to try and make sure the browser failed event came in before the exception (assuming there is an actual app crash behind this). If not an app crash, then we'd need the stack or a dump to try and tease out why it's throwing that error.

**Comment by RickStrahl** [comment]
Yeah firing ProcessFailed before the exception would definitely be better, but that doesn't appear what's happening. I log both and I don't actually see the ProcessFailed messages in my log at least not around the same time as the other failures. When these failures occur there are usually like 20-30 that occur in the same second.

My Web process fail handler stops processing and shows a dialog that the app will quit. But it looks to me that this never fires as I log before I get there and I don't see those in my logs.

Capture `InvalidArgumentException` is problematic because the handler that catches it is global. There are a lot of things that could throw this. At that point the only way I can check for this is by parsing the error message which is problematic with localization etc.

**Comment by champnic** [comment]
I've added this scenario to our backlog to revisit the custom exception discussion.

**Comment by RickStrahl** [comment]
To see if this helps at all I've made some code changes in my global error handler to trap for the error and immediately exit without even throwing up an error dialog which honestly only puts lipstick on this pig anyway :smile:

**Comment by champnic** [comment]
Hey @RickStrahl - I just realized you may be able to check for the actual HR code in the innerException, as a way to not have to compare against a string, for example:
`if (ex.InnerException.HResult == -2147019873)  // 0x8007139F: The group or resource is not in the correct state to perform the requested operation.`

We've also just put in a change to avoid some of the causes of this exception, especially the noisy ones due to layout updating that can't be avoided. Unfortunately I think it just missed the cuttoff for the recent prerelease SDK, but should be in the next one.
